src/duckling/game/collision/collision-size.ts

```diff
@@ -0,0 +1,31 @@
+import {Vector, vectorAbsolute, vectorSign, vectorMultiply, vectorDivide} from '../../math/vector';
+import {Entity} from '../../entitysystem/entity';
+import {immutableAssign} from '../../util/model';
+import {AssetService} from '../../project/asset.service';
+
+import {CollisionAttribute, getCollision} from './collision-attribute';
+import {collisionBoundingBox} from './collision-box';
+
+export function setCollisionSize(entity : Entity, newSize : Vector, assetService : AssetService) : CollisionAttribute {
```

Comment:
This and the other set functions are good candidates for unit testing. They are complex and pure.